CMakeLists.txt

```diff
@@ -332,12 +332,62 @@ add_custom_command(                       OUTPUT  ${syscall_macros_h}
   DEPENDS ${PROJECT_SOURCE_DIR}/scripts/gen_syscall_header.py
   )
 
-# This command is a hack to support commands that are always run. Any
-# target that depends on always_rebuild will always be rebuilt.
-add_custom_command(OUTPUT always_rebuild COMMAND cmake -E echo Building for board ${BOARD})
 
-set(syscall_list_h ${CMAKE_CURRENT_BINARY_DIR}/include/generated/syscall_list.h)
-set(syscalls_json  ${CMAKE_CURRENT_BINARY_DIR}/misc/generated/syscalls.json)
+set(syscall_list_h   ${CMAKE_CURRENT_BINARY_DIR}/include/generated/syscall_list.h)
+set(syscalls_json    ${CMAKE_CURRENT_BINARY_DIR}/misc/generated/syscalls.json)
+set(syscalls_subdirs ${CMAKE_CURRENT_BINARY_DIR}/misc/generated/syscalls_subdirs.txt)
+set(syscalls_trigger ${CMAKE_CURRENT_BINARY_DIR}/misc/generated/syscalls_subdirs.trigger)
```

Comment:
Could you add documentation somewhere that explains why syscalls_subdirs.txt could not be used as the trigger file?

Reply:
will do.

Currently I can inform that the reason is the difference between Linux and windows.
On windows, cmake will be re-invoked just when folder is updated.
However, on Linux we need to differentiate  between a modified file in existing folder (this triggers subdir to have its timestamp updated, and thus json re-run), whereas an added / removed folder triggers the trigger file, which causes cmake to re-run.

It is to aoid havingwith having two custom_commands depending on ${PARSE_SYSCALLS_PATHS_DEPENDS} which can make other strange behavior.
An option can be to make an add_custom_target to which multiple other targets can depend